### PR TITLE
Make indexer ready for inheritance classes

### PR DESCRIPTION
**Pull request by bhiemenz**

### CODE REVIEW

File: georocket-server/src/main/java/io/georocket/index/xml/BoundingBoxIndexer.java [hunk 1]
```diff
@@ -305,17 +305,11 @@ private boolean validate() {
   
   @Override
   public Map<String, Object> getResult() {
-    if (!boundingBoxInitialized) {
```
Comment (michel-kraemer) [inline georocket-server/src/main/java/io/georocket/index/xml/BoundingBoxIndexer.java]: I'm not sure if it is necessary to split this method. Wouldn't it be possible to just override `getResult`, call `super.getResult()` and then customise the result? At least, `checkResult` shouldn't log or change `boundingBoxInitialized`. These are side effects and the name `checkResult` implies it's free of side effects. In addition, `boundingBoxInitialized` really only makes sense in `getResult`.

**Comment by michel-kraemer** [comment]
Looks almost good. Thanks.

File: georocket-server/src/main/java/io/georocket/index/xml/BoundingBoxIndexer.java (hunk 1 above)
Comment (bhiemenz) [inline georocket-server/src/main/java/io/georocket/index/xml/BoundingBoxIndexer.java]: It is possible, yes. I reversed the bbox changes.

**Comment by michel-kraemer** [comment]
Perfect. Thanks!

**Pull request merged**